### ISSUE TITLE
Refactor PlotEmbeddings callback: Add proper label names and colormap support

### ISSUE THREAD

**Issue by MattScicluna** [latent-reasoning-works/manylatents#195]
# Summary

The `PlotEmbeddings` callback in https://github.com/latent-reasoning-works/manylatents/blob/main/manylatents/callbacks/embedding/plot_embeddings.py has accumulated technical debt that makes it difficult to support datasets with categorical labels (strings) alongside the numeric label indices required by `torch.tensor()` in `experiment.py`.

---

# Current Problem

When running experiments with categorical labels (e.g., population names like `"Yoruba"`, `"Han"`), there's a conflict between:

1. **experiment.py:518** — Converts labels to tensor for supervised learning:

```python
 train_labels = torch.tensor(labels)  # Fails for string labels
```

2.	experiment.py:543 — Passes labels to callbacks for visualization:
 ```python
"label": datamodule.test_dataset.get_labels()
```

3.	PlotEmbeddings — Uses labels for colormap-based coloring:
 ```python
ax = scprep.plot.scatter2d(..., cmap=cmap, c=color_array, ...)
```

For categorical data, scprep.plot.scatter2d expects string labels to match dict colormap keys like:
 ```python
{"Yoruba": "#FF0000", "Han": "#00FF00"}
```

But torch.tensor() requires integer labels.

# Technical Debt in PlotEmbeddings

1. No separation between label indices and label names
* _get_color_array() assumes labels can be used directly for both tensor conversion and colormap matching
* No get_label_names() convention exists in the dataset interface
* Datasets return either:
* strings (breaks tensor conversion), or
* integers (breaks colormap matching)

2. Ad-hoc colormap handling
* Colormaps are obtained via _get_colormap(dataset) which requires isinstance checks:
 ```python
if isinstance(dataset, DLAtree):
    cmap = cmap_dla_tree
else:
    cmap = "viridis"
```

* No standardized interface for datasets to provide their own colormap
* Downstream packages (manylatents-omics) must override _get_colormap() with more isinstance checks

3. Magic [1:] skip for dummy rows
* Both _get_embeddings() and _get_color_array() skip the first element. This was a workaround that is no longer necessary.
 ```python
return color_array[1:]
```

# Proposed Interface Improvements

## 1. Standardized label interface in datasets
```python
class Dataset:
    def get_labels(self) -> np.ndarray:
        """Return integer-encoded labels for tensor compatibility."""

    def get_label_names(self) -> np.ndarray:
        """Return string labels for visualization."""

    def classes_(self) -> np.ndarray:
        """Return unique label classes in sorted order."""
```

## 2. Standardized colormap interface
```python
class Dataset:
    def get_colormap(self) -> Optional[Dict[str, str]]:
        """Return colormap dict {label_name: hex_color} or None."""
```

## 4. Alternatively, do not change dataset but add `label_names` alongside labels in embeddings dict
```python
embeddings = {
    "embeddings": latents,
    "label": encoded_labels,       # For tensor operations
    "label_names": string_labels,  # For visualization
    "colormap": colormap,          # Optional dataset colormap
}
```
